admin-base/ui.apps/src/main/js/stateActions/selectAsset.js

```diff
@@ -34,9 +34,11 @@ export default function(me, target) {
     let view = me.getView()
     return new Promise( (resolve, reject) => {
         me.getApi().populateComponentDefinitionFromNode('/apps/admin/components/assetview')
-        .then( () => {
-            set(view, '/state/tools/asset/show', target.selected)
-            resolve()
+        .then(() => {
+            me.getApi().populateReferencedBy(target.selected).then( () => {
+                set(view, '/state/tools/asset/show', target.selected)
+                resolve()
+            }).catch( error => reject(error))
```

Comment:
Please take full advantage of the promise chaning by using returns and not nesting `then()` into each other.
Also, probably usefull to return the `target.selected` in the `resolve`. who knows, maybe we use it later
```js
.then(() => {
    return me.getApi().populateReferencedBy(target.selected)
})
.then( () => {
    set(view, '/state/tools/asset/show', target.selected)
    resolve(target.selected)
})
.catch( error => reject(error))
```

Reply:
Done